staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/connection.go

```diff
@@ -218,16 +218,12 @@ func (connection *VSphereConnection) NewClient(ctx context.Context) (*vim25.Clie
 		connection.RoundTripperCount = RoundTripperDefaultCount
 	}
 	client.RoundTripper = vim25.Retry(client.RoundTripper, vim25.TemporaryNetworkError(int(connection.RoundTripperCount)))
-	vcdeprecated, err := isvCenterDeprecated(client.ServiceContent.About.Version, client.ServiceContent.About.ApiVersion)
+	vcNotSupported, err := isvCenterNotSupported(client.ServiceContent.About.Version, client.ServiceContent.About.ApiVersion)
 	if err != nil {
-		klog.Errorf("failed to check if vCenter version:%v and api version: %s is deprecated. Error: %v", client.ServiceContent.About.Version, client.ServiceContent.About.ApiVersion, err)
-	}
-	if vcdeprecated {
-		// After this deprecation, vSphere 6.5 support period is extended to October 15, 2022 as
-		// https://blogs.vmware.com/vsphere/2021/03/announcing-limited-extension-of-vmware-vsphere-6-5-general-support-period.html
-		// In addition, the external vSphere cloud provider does not support vSphere 6.5.
-		// Please keep vSphere 6.5 support til the period.
-		klog.Warningf("vCenter is deprecated. version: %s, api verson: %s Please consider upgrading vCenter and ESXi servers to 6.7u3 or higher", client.ServiceContent.About.Version, client.ServiceContent.About.ApiVersion)
+		klog.Errorf("failed to check if vCenter version:%v and api version: %s is supported. Error: %v", client.ServiceContent.About.Version, client.ServiceContent.About.ApiVersion, err)
+	}
+	if vcNotSupported {
+		klog.Warningf("vCenter version is not supported. version: %s, api verson: %s Please consider upgrading vCenter and ESXi servers to 7.0u2 or higher", client.ServiceContent.About.Version, client.ServiceContent.About.ApiVersion)
```

Comment:
Thanks for making this change. I wonder about changing the messages in the logs though. 

One can still disable a beta feature gate and keep old behavior at least for one release.   Users who are upgrading to 1.25 without manually disabling feature gate will obviously run into problems.

But while the feature is in beta - technically it is still possible to disable the migration (even though it is enabled by default) and hence I wonder if Warning message here should additionally check if VSphere migration feature gate is enabled or not.

Reply:
Regardless of whether migration is enabled or disabled, we want to send out a message that vSphere releases less than 7.0u2 are no longer supported for the in-tree vSphere plugin from k8s 1.25 release.

Here we are just printing a warning message and does not error out which means the user can continue to use the in-tree driver on vSphere releases less than 7.0u2.

Reply:
I think sending warning no matter migration is enabled or disabled seems fine. It would be good the let customers know about this. Once it is GA, they cannot disable it anymore.

Reply:
maybe mention some more details? 
"vCenter version is not supported with CSI migration which is enabled by default and will be GA in future release"?

Reply:
@jingxu97 updated message as below

> 		klog.Warningf("vCenter version (version: %q, api verson: %q) is not supported for CSI Migration. Please consider upgrading vCenter and ESXi servers to 7.0u2 or higher for migrating vSphere volumes to CSI.", client.ServiceContent.About.Version, client.ServiceContent.About.ApiVersion)